src/win/util.c

```diff
@@ -484,7 +484,7 @@ uint64_t uv__hrtime(double scale) {
    * performance counter interval, integer math could cause this computation
    * to overflow. Therefore we resort to floating point math.
    */
-  return (uint64_t) ((double) counter.QuadPart * hrtime_interval_ * scale);
+  return (uint64_t) (counter.QuadPart * (uint64_t) scale / hrtime_interval_);
```

Comment:
Also, just to be clear, I'm not confident this has anything to do with the timer test failure either. I only pointed this issue (FP math on HR timer ticks will cause loss of precision) out to @Fishrock123 in io.js IRC as the first thing I'd point out if someone came to me with a problem that looked like an HR timer ticking backwards.